src/Controller/UserController.php

```diff
@@ -610,4 +611,164 @@ public function removeAvatar(
         }
         return new JsonResponse(['message' => 'Avatar successfully deleted!'], Response::HTTP_OK);
     }
+
+    /**
+     * Set Project Manager role
+     * @Rest\Post("/{id}/set_pm", requirements={"id"="\d+"})
+     * @SWG\Post(
+     *     tags={"User"},
+     *     summary="Set Project Manager role.",
+     *     description="Set Project Manager role.",
+     *     operationId="setProjectManagerRole",
+     *     produces={"application/json"},
+     *     @SWG\Parameter(
+     *     description="ID of User to set role",
+     *     in="path",
+     *     name="id",
+     *     required=true,
+     *     type="integer",
+     *     )
+     * )
+     * @SWG\Response(
+     *     response=401,
+     *     description="Unauthorized.",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="code", type="integer", example=401),
+     *     @SWG\Property(property="message", type="string", example="JWT Token not found"),
+     *     )
+     * )
+     * @SWG\Response(
+     *     response="200",
+     *     description="Successfull operation!",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="message", type="string", example="Role successfully set up!"),
+     *     )
+     * )
+     * @SWG\Response(
+     *     response="403",
+     *     description="Forbidden",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="code", type="integer", example=403),
+     *     @SWG\Property(property="message", type="string", example="Access denied!"),
+     *     )
+     * )
+     * @SWG\Response(
+     *     response=404,
+     *     description="Not found",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="code", type="integer", example=404),
+     *     @SWG\Property(property="message", type="string", example="Not found!"),
+     *     )
+     * )
+     * @param User $user
+     * @param UserRepository $userRepository
+     * @return JsonResponse
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function setPM(User $user, UserRepository $userRepository): JsonResponse
+    {
+        $hasAccess = $this->isGranted('ROLE_ADMIN');
+        if (!$hasAccess) {
+            return new JsonResponse([
+                'code' => Response::HTTP_FORBIDDEN,
+                'message' => 'Access denied!'
+            ], Response::HTTP_FORBIDDEN);
+        }
+        $user->setRoles(array('ROLE_PM'));
+        $userRepository->save($user);
+        return new JsonResponse(['message' => 'Role successfully set up!'], Response::HTTP_OK);
+    }
+
+    /**
+     * Set Project Manager
+     * @Rest\Post("/{userId}/set_pm/{pmId}", requirements={"userId"="\d+", "pmId"="\d+"})
```

Comment:
Rename `set_pm` to `assign-project-manager`. The ID of PM should be present in the body not in the URL.